### ISSUE TITLE
cilium cli install error about iptables

### ISSUE THREAD

**Issue by clcc2019** [cilium/cilium#20963]
### Is there an existing issue for this?

- [ ] I have searched the existing issues

### What happened?
```yaml
level=info msg="regenerating all endpoints" reason="one or more identities created or deleted" subsys=endpoint-manager
level=warning msg="Failed to install iptables rules" error="unable to run 'iptables -t raw -D OLD_CILIUM_OUTPUT_raw -o lxc+ -m mark --mark 0xa00/0xfffffeff -m comment --comment cilium: NOTRACK for proxy return traffic -j NOTRACK' iptables command: iptables: Bad rule (does a matching rule exist in that chain?).\n (exit status 1)" subsys=iptables
```
### Cilium Version

latest

### Kernel Version

5.11

### Kubernetes Version

1.22

### Code of Conduct

- [ ] I agree to follow this project's Code of Conduct

**Comment by aanm** [comment]
@clcc2019 which cilium version are you using?

**Comment by clcc2019** [comment]
> @clcc2019 which cilium version are you using?

```yaml
cilium-cli: v0.12.1 compiled with go1.18.5 on darwin/arm64
cilium image (default): v1.12.0
```

**Comment by aanm** [comment]
@clcc2019 That has been fixed by https://github.com/cilium/cilium/pull/20885 which is available in v1.12.1 and will be available in cilium-cli v0.12.2.

**Comment by T-Kukawka** [comment]
@aanm this issue affects CNCF conformance tests that provide certification for the running k8s clusters. Do you have any estimate? 1.12.1. is crashing as well: https://github.com/cilium/cilium/issues/20963

**Comment by aanm** [comment]
@T-Kukawka are you hitting this error with 1.21.1?

**Comment by T-Kukawka** [comment]
@aanm yes indeed :)

**Comment by jibi** [comment]
Hey @T-Kukawka :wave: 

Which command are you using to install the `1.12.1` version? This bug should be fixed in that version, so I'd like to try reproduce the failure you are seeing

**Comment by T-Kukawka** [comment]
Sooo  as i wrote in slack we have faced an issue with 1.11.x (It seems a known problem in cilium when using kube-proxy: [cilium/cilium#16163](https://github.com/cilium/cilium/issues/16163)) while running CNCF tests.
Bug was fixed in 1.12.1. We upgraded the app to latest 1.21.1 and it was crashing. we found this issue that seemed  like the same basing on the error message [cilium/cilium#20963](https://github.com/cilium/cilium/issues/20963)

We have now decided to postpone the release with Cilium -  when we are done we can reproduce it again
